### PR TITLE
Fix 756 src/wayland/shm/pool.rs

### PR DESCRIPTION
**Pull request by TornaxO7**
Adding docs and unsafe wrapping stuff.
Needed for https://github.com/Smithay/smithay/issues/946.

### CODE REVIEW

**Comment by i509VCB** [comment]
You might want to squash down the 8 or 9 commits into one once you have it building

**Comment by TornaxO7** [comment]
okeydokey

File: src/wayland/shm/pool.rs [hunk 1]
```diff
@@ -17,8 +17,10 @@ use tracing::{debug, instrument, trace};
 
 thread_local!(static SIGBUS_GUARD: Cell<(*const MemMap, bool)> = Cell::new((ptr::null_mut(), false)));
```
Comment (i509VCB) [inline src/wayland/shm/pool.rs:18]: This file seems to be done, so mark it with the unsafe_op_in_unsafe_fn attribute.

Comment (TornaxO7) [reply src/wayland/shm/pool.rs:18]: Hm... I though that I should fix them step by step until all points have been fixed [here](https://github.com/Smithay/smithay/issues/946), so I'll add `unsafe_op_in_unsafe_fn` to `lib.rs`.

Comment (TornaxO7) [reply src/wayland/shm/pool.rs:18]: Hm... I though that I should fix them step by step until all points have been fixed https://github.com/Smithay/smithay/issues/946, so I'll add unsafe_op_in_unsafe_fn to lib.rs.

Comment (i509VCB) [reply src/wayland/shm/pool.rs:18]: Not to the lib.rs but this file (`src/wayland/shm/pool.rs`)

Comment (TornaxO7) [reply src/wayland/shm/pool.rs:18]: ok

**Review by i509VCB — APPROVED** [review]

**Pull request merged**